airgun/widgets.py

```diff
@@ -1322,9 +1406,40 @@ def read(self):
         if not self.has_rows:
             self.logger.debug('Table {} is empty'.format(self.locator))
             return []
-
+        if self.pagination.is_displayed:
+            return self._read_all()
         return super(SatTable, self).read()
 
+    def _read_all(self):
+        """Return all available table values with using pagination navigation."""
+        table_rows = []
+        page_number = 1
+        if self.pagination.current_page != page_number:
```

Comment:
thinking to go on first page no matter what, but probably that will not decrease code much

Reply:
The first page button can disabled if already on first page